### ISSUE TITLE
Getting MNIST image classifier accuracy, precision and other eval.stats() metrics as NaN !!!!

### ISSUE THREAD

**Issue by smilebuddy** [deeplearning4j/deeplearning4j#4099]
#### Issue Description

I am trying to build image classifier using deeplearning4J in spark cluster. Java code is attached. My code is running successfully over spark cluster using spark-submit command, but the my model evaluation result gives Accuracy as NaN. I earlier thought that it's an issue with the model parameter and i tried tuning hyper-parameters like learning rate(tried 0.01, 0.0001, 0.000001) but gor no success.
Please tell me what's wrong here?

#### Version Information

Please indicate relevant versions, including, if relevant:

Deeplearning4j version: 0.9.1
platform information (OS, etc) : linux, spark over hadoop cluster

[MnistMLPDistributedExample.txt](https://github.com/deeplearning4j/deeplearning4j/files/1323718/MnistMLPDistributedExample.txt)

**Comment by AlexDBlack** [comment]
Looks like this has already been fixed on master. I've added a test to confirm this: https://github.com/deeplearning4j/deeplearning4j/pull/4109 (that fails on 0.9.1).

Work-around for 0.9.1: for MNIST (10 outputs) and 64 evaluation batch size: replace the first line with the second.
```
//        Evaluation evaluation = sparkNet.evaluate(testData);
        Evaluation evaluation = sparkNet.doEvaluation(testData, 64, new Evaluation(10))[0];
```
